src/postgraphile/http/subscriptions.ts

```diff
@@ -186,134 +191,262 @@ export async function enhanceHttpServerWithSubscriptions<
-              params,
-              socket,
-              options,
-            })
-          : params;
-        const finalParams: typeof hookedParams & { query: DocumentNode } = {
-          ...hookedParams,
-          query:
-            typeof hookedParams.query !== 'string' ? hookedParams.query : parse(hookedParams.query),
-        };
-
-        // You are strongly encouraged to use
-        // `postgraphile:validationRules:static` if possible - you should
-        // only use this one if you need access to variables.
-        const moreValidationRules = pluginHook('postgraphile:validationRules', [], {
-          options,
-          req,
-          res,
-          variables: params.variables,
-          operationName: params.operationName,
-          meta,
-        });
-        if (moreValidationRules.length) {
-          const validationErrors: ReadonlyArray<GraphQLError> = validate(
-            params.schema,
-            finalParams.query,
-            moreValidationRules,
-          );
-          if (validationErrors.length) {
-            const error = new Error(
-              'Query validation failed: \n' + validationErrors.map(e => e.message).join('\n'),
+            let normalizedConnectionParams = {};
+            if (connectionParams) {
+              normalizedConnectionParams = lowerCaseKeys(connectionParams);
+              request['connectionParams'] = connectionParams;
+              request['normalizedConnectionParams'] = normalizedConnectionParams;
+            }
+
+            if (!request.headers.authorization && normalizedConnectionParams['authorization']) {
+              /*
+               * Enable JWT support through connectionParams.
+               *
+               * For other headers you'll need to do this yourself for security
+               * reasons (e.g. we don't want to allow overriding of Origin /
+               * Referer / etc)
+               */
+              request.headers.authorization = String(normalizedConnectionParams['authorization']);
+            }
+
+            socket['postgraphileHeaders'] = {
+              ...normalizedConnectionParams,
+              // The original headers must win (for security)
+              ...request.headers,
+            };
+          },
+          async onSubscribe(ctx, msg) {
+            const context = await getContext(ctx.socket, msg.id);
+
+            // Override schema (for --watch)
+            const schema = await getGraphQLSchema();
+
+            const { payload } = msg;
+            const args: ExecutionArgs = {
+              schema,
+              contextValue: context,
+              operationName: payload.operationName,
+              document: parse(payload.query),
+              variableValues: payload.variables,
+            };
+
+            // for supplying custom execution arguments
+            const hookedArgs = pluginHook
+              ? pluginHook('postgraphile:ws:onSubscribe', args, {
+                  ...ctx,
```

Comment:
Lets pass `ctx` directly rather than splatting it. That allows us to add things in future without worrying that they'll override things that `ctx` contains.
```suggestion
                  ctx,
```